### ISSUE TITLE
The proxy does not work when running with TSL authentication in out of cluster mode

### ISSUE THREAD

**Issue by ludusrusso** [clastix/capsule-proxy#74]
When running on out of cluster mode, `bearerToken` is empty. In this case, we have to skip impersonification here

https://github.com/clastix/capsule-proxy/blob/4a7e4c60479f8c1b372d8e43bd28b6183c3e2a45/internal/webserver/webserver.go#L298-L303

@prometherion I do not understand exactly why this is working in cluster mode but not in out of cluster mode. In my understanding, when the proxy performs namespace filtering, it should skip completely end user impersonification.

**Comment by ludusrusso** [comment]
Ok, if I skip impersonification when the bearerToken is empty, I'm able to list namespaces BUT I'm also able to access namespaces with a user not allowed to access them. This is normal in my mind since in this case I'm doing `k get pods -n my-namespace` with the admin user authentication (coming from mTLS).

In my mind, this proxy should DO NOT perform impersonification only when accessing namespaces and nodes, and perform it otherwise. It seems I'm missing something! @prometherion  @bsctl any help?

**Comment by prometherion** [comment]
The issue we're facing is just happening during the out-of-cluster mode, since we're not injecting the bearer token from the Service Account mounted in the pod.

> this proxy should DO NOT perform impersonification only when accessing namespaces and nodes, and perform it otherwise

We can get it with a simple middleware since #62 has been merged.

**Comment by ludusrusso** [comment]
Ok, it seems that there cause of it is in this lines:

https://github.com/clastix/capsule-proxy/blob/4a7e4c60479f8c1b372d8e43bd28b6183c3e2a45/internal/webserver/webserver.go#L297

When you call the namespace and nodes endpoint with the proxy authorized using the bearerToken it generates and error `missing groups claim in JWT` which makes impersonification to be skipped. This does NOT happen when you call other endpoints!

Still need to investigate why this happens!

**Comment by ludusrusso** [comment]
The problem seems related to

https://github.com/clastix/capsule-proxy/blob/4a7e4c60479f8c1b372d8e43bd28b6183c3e2a45/internal/webserver/webserver.go#L130-L150

**Comment by prometherion** [comment]
> In my mind, this proxy should DO NOT perform impersonification only when accessing namespaces and nodes, and perform it otherwise

Let me rephrase it: the impersonification should be handled only when accessing the catch-all route (`/`).

I already have a PoC working both in out-of-cluster as well in in-cluster mode, going to push some code to start a discussion @ludusrusso.

In the meanwhile, thanks for reporting this! 💪🏻
